Core/Core/RichContentEditor/RichContentEditor.js

```diff
@@ -237,31 +236,37 @@ window.addEventListener('touchend', e => {
             img.classList.remove('editor-active')
         }
     }
-    if (e.target.nodeName.toLowerCase() === 'a') {
-        editor.currentEditingLink = e.target
-        webkit.messageHandlers.linkTap.postMessage({
-            text: e.target.textContent,
-            href: e.target.href
-        })
+    if (editor.currentEditingLink) {
+        webkit.messageHandlers.link.postMessage('')
         e.preventDefault()
     }
     if (!editor.isDragging && e.target.nodeName.toLowerCase() === 'html') {
         editor.focus()
     }
 })
 
-content.addEventListener('focus', () => {
-    webkit.messageHandlers.focus.postMessage()
-})
 content.addEventListener('blur', () => {
     if (editor.isEmpty()) { content.textContent = '' }
-    webkit.messageHandlers.blur.postMessage()
 })
 
 content.addEventListener('paste', e => {
-    webkit.messageHandlers.paste.postMessage()
```

Comment:
no sense in sending an event up to native side, when all the info we need is already in this event.